pkg/upload/android-proguard.go

```diff
@@ -1,128 +1,150 @@
+
+			if !utils.FileExists(mappingPath) {
+				return fmt.Errorf("unable to find the merged_native_libs in " + path)
 			}
-			//	Path is a file
-		} else {
 
-			if configuration == "" {
-				log.Warn("`--configuration` missing from options for " + path)
-				log.Info("Skipping " + path)
-				continue
+			if variant == "" {
+				variant, err = android.GetVariant(mappingPath)
+
+				if err != nil {
+					return err
+				}
 			}
 
+			mappingFile = filepath.Join(mappingPath, variant, "mapping.txt")
+
 			if appManifestPath == "" {
-				log.Warn("`--app-manifest-path` missing from options for " + path)
-				log.Info("Skipping " + path)
-				continue
+				//	Get the expected path to the manifest using variant name from the given path
+				appManifestPath = filepath.Join(path, "app", "build", "intermediates", "merged_manifests", variant, "AndroidManifest.xml")
 			}
+		} else if filepath.Base(path) == "mapping.txt" {
+			mappingFile = path
+
+			if appManifestPath == "" {
+				if variant == "" {
+					//	Set the mergeNativeLibPath based off the file location e.g. outputs/mapping/<variant>/mapping.txt
+					mergedManifestPath := filepath.Join(path, "..", "..", "..", "..", "intermediates", "merged_manifests")
+
+					if filepath.Base(mergedManifestPath) == "merged_manifests" {
+						variant, err = android.GetVariant(mergedManifestPath)
 
-			if filepath.Base(path) == "mapping.txt" {
-				uploadFileOptions[configuration] = map[string]string{}
-				uploadFileOptions[configuration]["androidManifestPath"] = appManifestPath
-				uploadFileOptions[configuration]["mappingPath"] = path
-			} else {
-				log.Error(path+" is not a supported file. Please use `mapping.txt`", 1)
+						log.Info(mergedManifestPath)
+						log.Info(variant)
+
+						if err == nil {
+							appManifestPath = filepath.Join(mergedManifestPath, variant, "AndroidManifest.xml")
+						}
+					}
+				}
 			}
-		}
-	}
 
-	numberOfVariants := len(uploadFileOptions)
+		}
 
-	if numberOfVariants < 1 {
-		log.Info("No variants to process")
-		return nil
-	}
+		// Check to see if we need to read the manifest file due to missing options
+		if apiKey == "" || applicationId == "" || buildUuid == "" || versionCode == "" || versionName == "" {
 
-	log.Info("Processing " + strconv.Itoa(numberOfVariants) + " variant(s)")
+			if variant == "" {
+				return fmt.Errorf("missing variant. Please specify using `--variant`")
+			}
 
-	for variant, config := range uploadFileOptions {
-		log.Info("Processing mapping.txt for variant: " + variant)
+			log.Info("Reading data from AndroidManifest.xml")
+			manifestData, err := android.ParseAndroidManifestXML(appManifestPath)
 
-		log.Info("Gathering information from AndroidManifest.xml")
+			if err != nil {
+				return err
+			}
 
-		androidManifestData, err := android.ParseAndroidManifestXML(config["androidManifestPath"])
+			if apiKey == "" {
+				log.Info("Setting API key from AndroidManifest.xml")
```

Comment:
Did we only want to do this if we're pulling it from the manifest? I added logging to the `BuildAndroidProguardUploadOptions` function so that if they're set no matter where it comes from it outputs them.